src/aof.c

```diff
@@ -146,7 +167,7 @@ void aofRewriteBufferAppend(unsigned char *s, unsigned long len) {
             int numblocks;
 
             block = zmalloc(sizeof(*block));
-            block->free = AOF_RW_BUF_BLOCK_SIZE;
+            block->free = zmalloc_usable_size(block)-offsetof(aofrwblock,buf);
```

Comment:
@ShooterIT i think i may have suggested the approach you use here in #8966 because there `repl_backlog_size` wasn't of type size_t (it was `long long`) and you did a casting that could have blown up in 32bit builds.
the code there was one that's executed very few times, so the concern of performance (calling malloc_size twice was not a concern), but here it might (although really minor).

so i wanna suggest to use `zmalloc_usable`, have some stack based size_t variable to later be stored into `block->free`. 
WDYT?

Reply:
OK, let me do it

Reply:
ok, let me do it

Reply:
done, we may encounter heavy write traffic, and that doesn't bring any trouble

Reply:
For this point, we also should  use `zmalloc_usable` in this line https://github.com/redis/redis/blob/unstable/src/networking.c#L328

Reply:
yes, good idea. there the impact might be higher (although i suppose still really minor).
care to fix that as part of one of your related PRs?

Reply:
Done in #8968